x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/components/analytics_tables/inline_tags_list.tsx

```diff
@@ -14,33 +14,39 @@ import { Query } from '../../types';
 
 interface Props {
   tags?: Query['tags'];
+  displayCountOnly?: boolean;
 }
-export const InlineTagsList: React.FC<Props> = ({ tags }) => {
+export const InlineTagsList: React.FC<Props> = ({ displayCountOnly, tags }) => {
   if (!tags?.length) return null;
 
   const displayedTags = tags.slice(0, 2);
-  const tooltipTags = tags.slice(2);
+  const tooltipTags = tags.slice(displayCountOnly ? 0 : 2);
+  const tagsText = tags.length === 1 ? 'tag' : 'tags';
+  const moreTagsMessage = displayCountOnly
+    ? `${tooltipTags.length} ${tagsText}`
+    : `and ${tooltipTags.length} more`;
```

Comment:
See above comment. There's also a separate i18n issue here where  `tagsText` is not being localized - we should prefer to handle pluralization through i18n now (see https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#pluralization).

```suggestion
    ? i18n.translate('xpack.enterpriseSearch.appSearch.engine.analytics.table.tagsCountBadge', {
        defaultMessage: '{tagsCount, plural, one {# tag} other {# tags}}',
        values: { tagsCount: tags.length },
      })
    : i18n.translate('xpack.enterpriseSearch.appSearch.engine.analytics.table.moreTagsBadge', {
        defaultMessage: 'and {moreTagsCount} more',
        values: { moreTagsCount: tooltipTags.length },
      });
```

Feel free to ping me to explain more about the above code if it's confusing!

Reply:
https://github.com/elastic/kibana/pull/92939/commits/b3a8b4f9eaf881965defb7bdcfea01443adc9562